static/src/javascripts/projects/common/modules/identity/upsell/store/consents.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+import {
+    getAllConsents,
+    getUserFromApi,
+    setConsent,
+} from 'common/modules/identity/api';
+
+type Consent = {
+    id: string,
+    name: string,
+    description: string,
+    isOptOut: boolean,
+    isChannel: boolean,
+    hasConsented: ?boolean,
+};
+
+const getUserConsents = (): Promise<string[]> =>
+    new Promise(accept => {
+        getUserFromApi(user => {
+            if (user && user.consents) {
+                accept(
+                    user.consents
+                        .filter(consent => consent.consented === true)
+                        .map(consent => consent.id)
+                );
+            } else {
+                accept([]);
+            }
+        });
+    });
+
+const fetchConsents = Promise.all([getUserConsents(), getAllConsents()]);
+
+const get = (): Promise<Consent[]> =>
+    fetchConsents.then(([acceptedConsents, allConsents]) =>
+        allConsents.map(c => ({
+            ...c,
+            hasConsented: acceptedConsents.includes(c.id),
+        }))
+    );
+
+const updateRemotely = (
```

Comment:
is this necessary? Could we just use `setConsent` directly instead? Feel free to swap the arg order of `setConsent` if that's what you were after...

Reply:
this allows us to centralise "committing" the changes remotely. This needs to be slightly abstracted as the opt-outs for example are sent as a group instead of when clicking them, so the state of the rendered consents will be intentionally inconsistent with the actual server-side consents until the user clicks 'save'